dask/dataframe/ewm.py

```diff
@@ -0,0 +1,112 @@
+from ..base import tokenize
+from ..utils import funcname
+from ..highlevelgraph import HighLevelGraph
+from .core import _emulate
+from .utils import make_meta
+
+
+def map_ewm_merge(func, merge_func, df, *args, **kwargs):
+    """Apply an ewm-type function to the dataframe, merging the results of applying
+       the function to an increasing prefix of partitions.
+
+    Parameters
+    ----------
+    func : function
+        Function applied to the dataframe.
+    df : dd.DataFrame, dd.Series
+    merge_func: function
+        Function that merges the partial result of applying func to the current partition
+        with results from previous partitions.
+    args, kwargs :
+        Arguments and keywords to pass to the function. The partition will
+        be the first argument, and these will be passed *after*.
+    """
+
+    merge_func_name = funcname(merge_func)
+    if "token" in kwargs:
+        func_name = kwargs.pop("token")
+        token = tokenize(df, *args, **kwargs)
+    else:
+        func_name = "ewm-" + funcname(func)
+        token = tokenize(func, merge_func, df, *args, **kwargs)
+
+    if "meta" in kwargs:
+        meta = kwargs.pop("meta")
+    else:
+        meta = _emulate(func, df, *args, **kwargs)
+    meta = make_meta(meta, index=df._meta.index)
+
+    name = "{0}-{1}".format(func_name, token)
+    name_partial = "partial-" + func_name
+    name_merge = "merge-" + merge_func_name
+
+    dsk = {}
+
+    for i, input_key in enumerate(df.__dask_keys__()):
+        dsk.update({(name_partial, i): (func, input_key, args, kwargs)})
+        kwargs["min_periods"] = max(
+            0, kwargs["min_periods"] - df.partitions[i].shape[kwargs["axis"]]
+        )
+
+    partial_keys = list(dsk)
+    dsk.update(
+        {
+            (name_merge, i): (
+                merge_func,
+                input_key,
+                partial_keys[:i],
+                df.__dask_keys__()[:i],
```

Comment:
To make sure I understand, this won't (can't) be computed in parallel right? The results from partition `4` depend on the result from partitions `0, 1, 2, 3`?

Reply:
And it's just the *result* of previous partitions that we need (at least for the functions you've done so far) Not the inputs?

Reply:
Yes, merging the results from partitions isn't parallelizable in this implementation. This is the cost of avoiding quadratic work (and linear time) that would be required to compute the EWM operations by definition.
And yes, that `input_key` does refer to a result from previous partitions, not the inputs. I will change the name to something that make more sense.